pkg/deploy/api/v1/conversion.go

```diff
@@ -143,24 +206,6 @@ func convert_v1_DeploymentTriggerImageChangeParams_To_api_DeploymentTriggerImage
 	if err := s.Convert(&in.From, &out.From, 0); err != nil {
 		return err
 	}
-	switch in.From.Kind {
```

Comment:
The kinds are still copied as part of the object reference. This part seems to be for populating the deprecated internal fields (RepositoryName and Tag).